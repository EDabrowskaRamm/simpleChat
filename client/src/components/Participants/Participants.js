import React, { useState, useEffect } from 'react'
import { connect } from 'react-redux'

import {
  ParticipantsRoot,
  ParticipantTile,
  ParticipantsWrapper
} from './Participants.shards'

import Input from '../Input/Input'

import { addParticipant, addUser } from '../../store/actions'

import { participantID } from '../../utils/helpers'
import { URL, EVENT_TYPE } from '../../utils/consts.js'

const ConnectedParticipants = ({
  getParticipants,
  addParticipant,
  addUser,
  getUser
}) => {
  const [newName, setNewName] = useState({
    name: '',
    id: participantID()
  })
  const ws = new WebSocket(URL)

  useEffect(() => {
    ws.onopen = () => {
      console.log('connected participant')
    }

    ws.onmessage = evt => {
      const socketData = JSON.parse(evt.data)

      if (socketData.type === EVENT_TYPE.USER_EVENT) {
        addParticipant(socketData)
      }
    }

    ws.onclose = () => {
      console.log('disconnected')
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [])

  const handleChange = event => {
    setNewName({
      name: event.target.value,
      id: newName.id
    })
  }

  const handleSubmit = event => {
    event.preventDefault()
    ws.send(JSON.stringify({
      ...newName,
      type: EVENT_TYPE.USER_EVENT
    }))

    addUser(newName)

    setNewName({
      name: '',
      id: ''
    })
  }

  return <ParticipantsRoot>
    <ParticipantsWrapper>
      {getParticipants.map(participant => <ParticipantTile
        key={participant.id}
      >
        <p>{participant.name}</p>
      </ParticipantTile>)}
    </ParticipantsWrapper>
    {!getUser.name && <Input
      handleSubmit={handleSubmit}
      id='name'
      value={newName.name}
      placeholder='Type in your name'
      handleChange={handleChange}
    />}
  </ParticipantsRoot>
}

const Participants = connect(
  null,
  dispatch => ({
    addParticipant: (participant) => dispatch(addParticipant(participant)),
    addUser: (user) => dispatch(addUser(user))
  })
)(ConnectedParticipants)

export default Participants
