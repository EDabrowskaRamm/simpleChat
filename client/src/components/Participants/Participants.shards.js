import styled from '@emotion/styled'

export const ParticipantsRoot = styled.section`
  width: 100%;
  height: calc(100vh - 240px);
  display: flex;
  flex-direction: column;
`
export const ParticipantsWrapper = styled.div`
  height: 100%;
  padding: 0 30px;
  overflow: auto;
  flex-grow: 1;
`

export const ParticipantTile = styled.div`
  display: flex;
  align-items: center;
  width: 100%;
  height: 120px;
  padding-left: 12px;
  border-bottom: ${({ theme }) => theme.commons.borders.main};
  text-transform: capitalize;
`