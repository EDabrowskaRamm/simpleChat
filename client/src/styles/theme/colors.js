const _white = '#FFF'
const _colorGrey56 = '#868c97'
const _colorGrey61 = '#979c9f'
const _colorGrey63 = '#9a9fa8'
const _colorGrey89 = '#e1e3e5'
const _colorGrey92 = '#ebebeb'

const _pictonBlue = '#3d90f1'
const _oxfordBlue = '#354052'

export const background = {
  primary: _white,
  secondary: _colorGrey92,
}

export const text = {
  primary: _oxfordBlue,
  secondary: _colorGrey63,
  link: _pictonBlue,
  placeholder: _colorGrey61,
  left: _colorGrey56,
}

export const border = {
  input: _colorGrey56,
  chat: _colorGrey89,
  transparent: 'transparent',
}
