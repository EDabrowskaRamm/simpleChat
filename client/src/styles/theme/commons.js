import { border } from './colors'

export const borders = {
  main: `1px solid ${border.chat}`,
  input: `1px solid ${border.input}`,
}
