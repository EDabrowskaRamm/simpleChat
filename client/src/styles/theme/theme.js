import * as colors from './colors'
import * as commons from './commons'

export default {
  colors,
  commons
}
