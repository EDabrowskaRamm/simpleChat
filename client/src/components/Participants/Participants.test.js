import React from 'react'
import renderer from 'react-test-renderer'
import serializer from 'jest-emotion'
import { ThemeProvider } from 'emotion-theming'

import Participants from './Participants'

import theme from '../../styles/theme/theme'

expect.addSnapshotSerializer(serializer)

describe('renders Participants', () => {
  it('Snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <Participants />
      </ThemeProvider>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
