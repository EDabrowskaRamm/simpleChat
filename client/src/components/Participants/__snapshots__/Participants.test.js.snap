// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Participants Snapshot 1`] = `
.emotion-3 {
  width: 100%;
  height: calc(100vh - 240px);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-2 {
  height: 100%;
  padding: 0 30px;
  overflow: auto;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 120px;
  padding-left: 12px;
  border-bottom: 1px solid #e1e3e5;
  text-transform: capitalize;
}

<section
  className="emotion-3"
>
  <div
    className="emotion-2"
  >
    <div
      className="emotion-0"
    >
      <p>
        Jane Downey
      </p>
    </div>
    <div
      className="emotion-0"
    >
      <p>
        michel sagan
      </p>
    </div>
  </div>
</section>
`;
